utility_network/perform-valve-isolation-trace/src/main/java/com/esri/samples/perform_valve_isolation_trace/PerformValveIsolationTraceController.java

```diff
@@ -0,0 +1,308 @@
+import com.esri.arcgisruntime.utilitynetworks.UtilityDomainNetwork;
+import com.esri.arcgisruntime.utilitynetworks.UtilityElement;
+import com.esri.arcgisruntime.utilitynetworks.UtilityElementTraceResult;
+import com.esri.arcgisruntime.utilitynetworks.UtilityNetwork;
+import com.esri.arcgisruntime.utilitynetworks.UtilityNetworkDefinition;
+import com.esri.arcgisruntime.utilitynetworks.UtilityNetworkSource;
+import com.esri.arcgisruntime.utilitynetworks.UtilityTier;
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceConfiguration;
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceFilter;
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceParameters;
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceResult;
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceType;
+
+public class PerformValveIsolationTraceController {
+
+  @FXML private MapView mapView;
+  @FXML private ProgressIndicator progressIndicator;
+  @FXML private ComboBox<UtilityCategory> categorySelectionComboBox;
+  @FXML private Button traceButton;
+  @FXML private Label statusLabel;
+  @FXML private CheckBox includeIsolatedFeaturesCheckbox;
+
+  private UtilityNetwork utilityNetwork;
+  private UtilityTraceConfiguration traceConfiguration;
+  private UtilityElement startingLocation;
+
+  public void initialize() {
+    try {
+
+      // create a basemap and set it to the map view
+      ArcGISMap map = new ArcGISMap(Basemap.createStreetsNightVector());
+      mapView.setMap(map);
+
+      // load the utility network data from the feature service and create feature layers
+      String featureServiceURL =
+              "https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleGas/FeatureServer";
+
+      ServiceFeatureTable distributionLineFeatureTable = new ServiceFeatureTable(featureServiceURL + "/3");
+      FeatureLayer distributionLineLayer = new FeatureLayer(distributionLineFeatureTable);
+
+      ServiceFeatureTable deviceFeatureTable = new ServiceFeatureTable(featureServiceURL + "/0");
+      FeatureLayer deviceLayer = new FeatureLayer(deviceFeatureTable);
+
+      // add the feature layers to the map
+      map.getOperationalLayers().addAll(Arrays.asList(distributionLineLayer, deviceLayer));
+
+      // create a graphics overlay for the starting location and add it to the map view
+      GraphicsOverlay startingLocationGraphicsOverlay = new GraphicsOverlay();
+      mapView.getGraphicsOverlays().add(startingLocationGraphicsOverlay);
+
+      // create and apply renderers for the starting point graphics overlay
+      SimpleMarkerSymbol startingPointSymbol =
+              new SimpleMarkerSymbol(SimpleMarkerSymbol.Style.CROSS, ColorUtil.colorToArgb(Color.LIGHTGREEN), 25);
+      startingLocationGraphicsOverlay.setRenderer(new SimpleRenderer(startingPointSymbol));
+
+      // create and load the utility network
+      utilityNetwork = new UtilityNetwork(featureServiceURL, map);
```

Comment:
looking at our doc for utility network, I'm not convinced this is the best sample to show this constructor. The doc says the following for the constructor that takes a URL and a map:

> Usually used when instantiating a UtilityNetwork object from a web map.

Since we aren't doing this, I think it's more appropriate here to use the basic constructor which just takes the URL, unless there was a specific reason you chose this constructor? if so a comment as to why is probably good.

Reply:
I didn't make any conscious choice here, but just followed the design. Anyway, I've made this use the regular constructor